app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -473,8 +473,8 @@ protected function getTaxForAdmin( $taxTotal ) {
     protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
     {
         // we'll check if we can simply dividing by two corrects the problem
-        if ($projectedTotal == (int)($magentoDerivedCartData['total_amount']/2)) {
-            $magentoDerivedCartData["total_amount"] = (int)($magentoDerivedCartData['total_amount']/2);
+        if ($projectedTotal == (int)round($magentoDerivedCartData['total_amount']/2)) {
+            $magentoDerivedCartData["total_amount"] = (int)round($magentoDerivedCartData['total_amount']/2);
```

Comment:
There is no proof that this is needed.  It falls outside of the scope of what you identified as a problem.  Please limit your changes to validation step.